### PR TITLE
Fix issue 715

### PR DESCRIPTION
**Pull request by xieydd**
### Ⅰ. Describe what this PR does
fix bug #715

### CODE REVIEW

**Comment by xieydd** [comment]
/hold 
Add test for allocatePorts function.

**Comment by xieydd** [comment]
@cheyang @TrafalgarZZZ Can you take a look at `ineffectual assignment to index` error

**Comment by cheyang** [comment]
```
pkg/ddc/alluxio/transform.go:405:3: ineffectual assignment to index (ineffassign)
		index += 2
		^
Error: Process completed with exit code 1.
```

File: pkg/ddc/alluxio/transform.go [hunk 1]
```diff
@@ -381,26 +381,28 @@ func (e *AlluxioEngine) allocatePorts(value *Alluxio) error {
...
 				"len(allocatedPorts)", len(allocatedPorts),
 				"allocatedPorts", allocatedPorts)
-			return fmt.Errorf("The lengh of port list is %v less than expected %v", len(allocatedPorts), PORT_NUM)
+			return fmt.Errorf("The lengh of port list is %v less than expected %v", len(allocatedPorts), index+2)
 		}
-		value.Master.Ports.Embedded = allocatedPorts[index+1]
-		value.Master.Ports.Embedded = allocatedPorts[index+2]
+		value.Master.Ports.Embedded = allocatedPorts[index]
+		value.JobMaster.Ports.Embedded = allocatedPorts[index+1]
+		index += 2
```
Comment (TrafalgarZZZ) [inline pkg/ddc/alluxio/transform.go:405]: `index += 2` seems no effect here？

Comment (xieydd) [reply pkg/ddc/alluxio/transform.go:405]: tks

**Pull request merged**